src/arrayMethodJoin.js

```diff
@@ -4,8 +4,29 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let result = '';
+
+    if (this.length === 0) {
+      return result;
+    }
+
+    for (let i = 0; i < this.length - 1; i++) {
+      if (typeof this[i] === 'undefined' || this[i] === null) {
+        this[i] = '';
+      }
+      result += this[i];
+      result += separator;
+    }
+
+    if (typeof this[this.length - 1] === 'undefined'
+    || this[this.length - 1] === null) {
+      this[this.length - 1] = '';
+    }
```

Comment:
Why you need this?

Reply:
I've made some corrections, but i need this to check the last element of the array, it might be null or undefined. Can I check it in another way?